src/sdk/nsfs_glacier_backend/tapecloud.js

```diff
@@ -21,11 +23,49 @@ function get_bin_path(bin_name) {
     return path.join(config.NSFS_GLACIER_TAPECLOUD_BIN_DIR, bin_name);
 }
 
-async function get_task(task_id) {
-    return await exec(`${get_bin_path(TASK_SHOW_SCRIPT)} ${task_id}`, { return_stdout: true });
+/**
+ * 
+ * @param {*} task_id 
+ * @param {(string) => Promise<void>} recorder 
+ */
+function record_failed_tasks(task_id, recorder) {
+    return new Promise((resolve, reject) => {
+        const proc = spawn(get_bin_path(TASK_SHOW_SCRIPT), [task_id]);
+        const reader = createInterface(proc.stdout);
+        reader.on('line', async line => {
+            if (!line.startsWith("Fail")) return;
+
+            const parsed = line.split(/\s+/);
+            if (parsed.length !== 6) {
+                reject(Error('failed to parse task show'));
+            }
+
+            if (parsed[1] !== ERROR_DUPLICATE_TASK) {
+                // Column 5 is the filename (refer tapecloud [eeadm] manual)
+                await recorder(parsed[5]);
+            }
+        });
+
+        let errcode = 0;
+
+        proc.on('exit', code => { errcode = code; });
+        reader.on('close', () => {
```

Comment:
As per the docs of readline:
> The input stream receives its 'end' event;

I expect that this event will occur only once we are done going over all the lines. @guymguym how fair do you think is this assumption?

Reply:
https://nodejs.org/docs/latest/api/readline.html#example-read-file-stream-line-by-line

reader.on('line', async ...) will not really wait for the internal promise - I think you need to use for await for that backpressure of the input stream, and will only finish the async iterator loop once the input stream ends.